challenges/2021_CyrilMorcrette_cloudBaseHeight/cbh_challenge_send_data_to_zarr.ipynb

```diff
@@ -0,0 +1,2569 @@
+{
```

Comment:
<pre class="ql-syntax" spellcheck="false">Line #5.    f = open(path_to_save_result, "r+b")
</pre><p>The pythonic recommendation is to use the "with" operator for file operation so files are automatically closed. This is called the "resource acquisition is initialization" or RAII paradigm (https://en.wikipedia.org/wiki/Resource_acquisition_is_initialization).</p><p> </p><p>The with operator uses special overloaded functions on python objects called <code>__enter__</code> and <code>__exit__</code>&nbsp;as described here https://docs.python.org/3/reference/compound_stmts.html#the-with-statement</p><p>In this case it would look like:</p><pre class="ql-syntax" spellcheck="false">with open(path_to_save_result, 'r+b') as f:
    numpy_files = np.load(f)
    # do other stuff

# no need to close, because the __exit__ has been called at the end of the with block.
</pre>
<br/>

 _Reply via <a href='https://app.reviewnb.com/MetOffice/data_science_cop/pull/30/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='42'/>